contrib/devtools/symbol-check.py

```diff
@@ -15,45 +15,55 @@
 
 import lief #type:ignore
 
-# Debian 9 (Stretch) EOL: 2022. https://wiki.debian.org/DebianReleases#Production_Releases
+# MAX_VERSIONS defines the maximum versions for dynamic symbols defined in linux
+# binaries for release. These are static for each major version of Dogecoin Core.
 #
-# - g++ version 6.3.0 (https://packages.debian.org/search?suite=stretch&arch=any&searchon=names&keywords=g%2B%2B)
-# - libc version 2.24 (https://packages.debian.org/search?suite=stretch&arch=any&searchon=names&keywords=libc6)
+# Debian 6.0.9 (Squeeze) has:
 #
-# Ubuntu 16.04 (Xenial) EOL: 2026. https://wiki.ubuntu.com/Releases
+# - g++ version 4.4.5 (https://packages.debian.org/search?suite=default&section=all&arch=any&searchon=names&keywords=g%2B%2B)
+# - libc version 2.11.3 (https://packages.debian.org/search?suite=default&section=all&arch=any&searchon=names&keywords=libc6)
+# - libstdc++ version 4.4.5 (https://packages.debian.org/search?suite=default&section=all&arch=any&searchon=names&keywords=libstdc%2B%2B6)
 #
-# - g++ version 5.3.1
-# - libc version 2.23
+# Ubuntu 10.04.4 (Lucid Lynx) has:
 #
-# CentOS Stream 8 EOL: 2024. https://wiki.centos.org/About/Product
+# - g++ version 4.4.3 (http://packages.ubuntu.com/search?keywords=g%2B%2B&searchon=names&suite=lucid&section=all)
+# - libc version 2.11.1 (http://packages.ubuntu.com/search?keywords=libc6&searchon=names&suite=lucid&section=all)
+# - libstdc++ version 4.4.3 (http://packages.ubuntu.com/search?suite=lucid&section=all&arch=any&keywords=libstdc%2B%2B&searchon=names)
 #
-# - g++ version 8.5.0 (http://mirror.centos.org/centos/8-stream/AppStream/x86_64/os/Packages/)
-# - libc version 2.28 (http://mirror.centos.org/centos/8-stream/AppStream/x86_64/os/Packages/)
+# Taking the minimum of these as our target.
 #
-# See https://gcc.gnu.org/onlinedocs/libstdc++/manual/abi.html for more info.
+# According to GNU ABI document (http://gcc.gnu.org/onlinedocs/libstdc++/manual/abi.html) this corresponds to:
+#   GCC 4.4.0: GCC_4.4.0
+#   GCC 4.4.2: GLIBCXX_3.4.13, CXXABI_1.3.3
+#   (glibc)    GLIBC_2_11
 
 MAX_VERSIONS = {
-'GCC':       (4,8,0),
+'GCC':       (4,4,0),
 'GLIBC': {
-    lief.ELF.ARCH.x86_64: (2,18),
-    lief.ELF.ARCH.ARM:    (2,18),
-    lief.ELF.ARCH.AARCH64:(2,18),
-    lief.ELF.ARCH.PPC64:  (2,18),
+    lief.ELF.ARCH.x86_64: (2,11),
+    lief.ELF.ARCH.i386:   (2,11),
+    lief.ELF.ARCH.ARM:    (2,11),
+    lief.ELF.ARCH.AARCH64:(2,17),
+    lief.ELF.ARCH.PPC64:  (2,17),
     lief.ELF.ARCH.RISCV:  (2,27),
 },
-'LIBATOMIC': (1,0),
-'V':         (0,5,0),  # xkb (bitcoin-qt only)
+'CXXABI':  (1,3,3),
+'GLIBCXX': (3,4,13),
+'V':         (0,5,0),  # xkb (dogecoin-qt only)
 }
 
 # Ignore symbols that are exported as part of every executable
 IGNORE_EXPORTS = {
-'environ', '_environ', '__environ', '_fini', '_init', 'stdin',
-'stdout', 'stderr',
+  '_edata', '_end', '_init', '__bss_start', '_fini', '_IO_stdin_used',
+  'stdin', 'stdout', 'stderr'
```

Comment:
Are `environ` and `_environ` no longer exported? Is this cleanup we should have done earlier?